### PR TITLE
[KYUUBI #1031] Implement api: /sessions/count

### PR DESCRIPTION
**Pull request by yanghua**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/src/test/scala/org/apache/kyuubi/server/RestFrontendServiceSuite.scala [hunk 1]
```diff
@@ -80,10 +83,41 @@ class RestFrontendServiceSuite extends KyuubiFunSuite{
     server.stop()
   }
 
-  class RestNoopServer extends NoopServer {
+  test("kyuubi rest frontend service sessions resource") {
+    val server = new RestNoopServer()
```
Comment (ulysses-you) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/server/RestFrontendServiceSuite.scala:87]: shall we add a closure and simplify to `withKyuubiRestServer {}`

File: kyuubi-server/src/test/scala/org/apache/kyuubi/server/RestFrontendServiceSuite.scala [hunk 2]
```diff
@@ -61,29 +64,62 @@ class RestFrontendServiceSuite extends KyuubiFunSuite{
...
+          assert(html.toLowerCase(Locale.ROOT).equalsIgnoreCase(expectedStr))
+        }
+    }
+  }
+
+  def withKyuubiRestServer(f: (RestNoopServer, RestFrontendService) => Unit): Unit = {
     val server = new RestNoopServer()
     server.stop()
     val conf = KyuubiConf()
     conf.set(KyuubiConf.FRONTEND_REST_BIND_HOST, "localhost")
```
Comment (ulysses-you) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/server/RestFrontendServiceSuite.scala:94]: we can also specify the port here

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/RestFrontendService.scala [hunk 3]
```diff
@@ -114,11 +114,13 @@ private[server] class RestFrontendService private(name: String, be: BackendServi
   private def stopHttpServer(): Unit = {
     if (jettyServer != null) {
       try {
+        connector.stop()
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/RestFrontendService.scala:117]: shall we do this in another try-catch ?

File: kyuubi-server/src/test/scala/org/apache/kyuubi/server/RestFrontendServiceSuite.scala (hunk 2 above)
Comment (yanghua) [reply kyuubi-server/src/test/scala/org/apache/kyuubi/server/RestFrontendServiceSuite.scala:94]: Here for testing, we can use the default port?

Comment (ulysses-you) [reply kyuubi-server/src/test/scala/org/apache/kyuubi/server/RestFrontendServiceSuite.scala:94]: We can return host:port in closure so we don't need hard code them. And the `server` seems not useful, seems `frontendService`  is enough. what do you thank ?

Comment (yanghua) [reply kyuubi-server/src/test/scala/org/apache/kyuubi/server/RestFrontendServiceSuite.scala:94]: reasonable, +1

File: dev/dependencyList [hunk 4]
```diff
@@ -33,6 +33,7 @@ htrace-core4/4.1.0-incubating//htrace-core4-4.1.0-incubating.jar
 jackson-annotations/2.11.4//jackson-annotations-2.11.4.jar
 jackson-core/2.11.4//jackson-core-2.11.4.jar
 jackson-databind/2.11.4//jackson-databind-2.11.4.jar
+jackson-module-jaxb-annotations/2.9.9//jackson-module-jaxb-annotations-2.9.9.jar
```
Comment (pan3793) [inline dev/dependencyList:36]: we should pin `jackson-module-jaxb-annotations:${jackson.version}` in root `pom.xml`

**Review by yaooqinn — APPROVED** [review]

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thank you @yanghua , merging to master

**Pull request closed without merging**